hub-remote-deploy

```diff
@@ -0,0 +1,196 @@
+configMapGenerator:
+- name: toolbox-env
+  namespace: $K8S_NAMESPACE
+  env: envfile
+- name: user-group
+  namespace: $K8S_NAMESPACE
+  literals:    
+  - USER=$USER
+  - UID=$(id -u)
+  - GID=$(id -g)
+resources:
+- deployment.yaml
+EOF
+
+pod_manifest="$KUSTOMIZE_DIR/deployment.yaml"
+cat <<EOF > $pod_manifest
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: toolbox
+  namespace: $K8S_NAMESPACE
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: toolbox-sa
+subjects:
+- kind: ServiceAccount
+  name: toolbox
+  namespace: $K8S_NAMESPACE
+roleRef:
+  kind: ClusterRole
+  name: cluster-admin
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: toolbox
+  namespace: $K8S_NAMESPACE
+spec:
+  serviceAccountName: toolbox
+  containers:
+  - name: toolbox
+    image: $IMAGE:$IMAGE_VERSION
+    command: ["bash", "-c", "hub extensions install && sleep 10000"]    
+    envFrom:
+        - configMapRef:
+            name: toolbox-env
+        - configMapRef:
+            name: user-group
+EOF
+}
+
+deploy_toolbox() {
+    $SILENT || printf '\033[1mDeploying toolbox container\n\033[0m'
+
+    kubectl kustomize ${KUSTOMIZE_DIR} > "${KUSTOMIZE_DIR}/toolbox.yaml"    
+    kubectl apply -f ${KUSTOMIZE_DIR}/toolbox.yaml
+    $SILENT || printf '\033[1mWaiting for toolbox to be deployed\033[0m'
+    while [[ $(kubectl -n $K8S_NAMESPACE get pods -l app=toolbox -l provider=agilestacks.com -o 'jsonpath={..status.conditions[?(@.type=="Ready")].status}') != "True" ]]; do 
+        $SILENT || printf '\033[1m.\033[0m' && sleep 1; 
+    done
+    $SILENT || printf '\033[1m\n\033[0m'
+}
+
+delete_toolbox() {
+    $SILENT || printf '\033[1mDeleting toolbox container\n\033[0m'
+    kubectl -n "$K8S_NAMESPACE" delete pod -l app=toolbox -l provider=agilestacks.com; 
+    kubectl -n "$K8S_NAMESPACE"  delete cm -l app=toolbox -l provider=agilestacks.com; 
+    kubectl -n "$K8S_NAMESPACE" delete clusterrolebindings.rbac.authorization.k8s.io -l app=toolbox -l provider=agilestacks.com; 
+    kubectl -n "$K8S_NAMESPACE" delete sa -l app=toolbox -l provider=agilestacks.com
+}
+
+
+
+upload_data() {
+    $SILENT || printf '\033[1mCopying Certified Stack data to toolbox\n\033[0m'
+    kubectl -n "$K8S_NAMESPACE" cp . toolbox:./
+    kubectl -n "$K8S_NAMESPACE" cp ~/.aws toolbox:/root/
+}
+
+
+deploy_stack() {
+    $SILENT || printf '\033[1mDeploying Certified Stack from toolbox\033[0m'
+    kubectl -n "$K8S_NAMESPACE" exec -it toolbox -- bash -c "\
+        rm -f .env &&\
+        ln -s .hub/env/$HUB_DOMAIN_NAME.env .env &&\
+        source .env &&\
+        hub ext stack elaborate &&\
+        hub ext stack deploy $@
+        "
+}
+
+
+GUIDE=${GUIDE:-true}
+SILENT=${SILENT:-false}
+IMAGE=${IMAGE:-agilestacks/toolbox}
+IMAGE_VERSION=${IMAGE_VERSION:-latest}
+K8S_NAMESPACE=${K8S_NAMESPACE:-kube-system}
+DELETE_TOOLBOX=${K8S_NAMESPACE:-true}
+KUSTOMIZE_DIR=$(mktemp -d)
```

Comment:
is it better to use STDIN for kubectl/kustomize ?

Reply:
kubectl inbuilt kustomzie can't do STDIN, then I have you separate tool.